x-pack/plugins/watcher/server/models/watch/threshold_watch/build_visualize_query.js

```diff
@@ -117,11 +108,22 @@ export function buildVisualizeQuery(watch, visualizeOptions) {
     termOrder,
   });
   const body = watchInput.search.request.body;
-  const dateAgg = buildDateAgg({
-    field: watch.timeField,
-    interval: visualizeOptions.interval,
-    timeZone: visualizeOptions.timezone,
-  });
+  const dateAgg = {
+    date_histogram: {
+      field: watch.timeField,
+      time_zone: visualizeOptions.timezone,
+      min_doc_count: 1,
+    },
+  };
+
+  if (kibanaVersion.major < 8) {
+    // In 7.x we use the deprecated "interval" in date_histogram agg
+    dateAgg.date_histogram.interval = visualizeOptions.interval;
+  } else {
+    // From 8.x we use the more precise "fixed_interval" or "calendar_interval"
+    const intervalType = getIntervalType(visualizeOptions.interval);
+    dateAgg.date_histogram[`${intervalType}_interval`] = visualizeOptions.interval;
```

Comment:
How would you feel about changing `getIntervalType` to return `calendar_interval` or `fixed_interval`? I think this change will make the code more explicit about its tight coupling with the ES API. As it stands, the tight coupling still exists but it's an implicit tightly coupling due to this string concatenation.